src/components/PlantingSites/editor/SiteBoundary.tsx

```diff
@@ -17,21 +18,21 @@ export type SiteBoundaryProps = {
   site: PlantingSite;
 };
 
+const featureSiteBoundary = (id: number, boundary?: MultiPolygon): FeatureCollection | undefined =>
+  !boundary
+    ? undefined
+    : {
+        type: 'FeatureCollection',
+        features: [toFeature(boundary, {}, id)],
+      };
 export default function SiteBoundary({ onChange, onValidate, site }: SiteBoundaryProps): JSX.Element {
   const [description, setDescription] = useState<Description[]>([]);
-  const [siteBoundary, setSiteBoundary] = useState<FeatureCollection | undefined>();
+  const [siteBoundary, setSiteBoundary, undo, redo] = useUndoRedoState<FeatureCollection | undefined>(
+    featureSiteBoundary(site.id, site.boundary)
```

Comment:
we want the first state in the undo/redo stack to be the initial boundary state.. hence not using a useEffect which would lead to an initial undefined state (enabling undo up to that point)